cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/CaReconciler.java

```diff
@@ -579,8 +578,21 @@ Future<Void> rollingUpdateForNewCaKey() {
                 false,
                 eventPublisher
         ).rollingRestart(pod -> {
-            LOGGER.debugCr(reconciliation, "Rolling Pod {} due to {}", pod.getMetadata().getName(), podRollReasons.getReasons());
-            return podRollReasons;
+            if (podRollReasons.getReasons().size() == 1 && podRollReasons.contains(RestartReason.CLUSTER_CA_CERT_KEY_REPLACED)) {
+                // Check if this pod has been rolled previously
+                int clusterCaKeyGeneration = clusterCa.keyGeneration();
+                int podClusterCaKeyGeneration = Annotations.intAnnotation(pod, Ca.ANNO_STRIMZI_IO_CLUSTER_CA_KEY_GENERATION, clusterCaKeyGeneration);
+                if (clusterCaKeyGeneration == podClusterCaKeyGeneration) {
+                    LOGGER.debugCr(reconciliation, "Not rolling Pod {} since the CA cert key generation is correct.", pod.getMetadata().getName());
+                    return RestartReasons.empty();
+                } else {
+                    LOGGER.debugCr(reconciliation, "Rolling Pod {} due to {}", pod.getMetadata().getName(), podRollReasons.getReasons());
+                    return podRollReasons;
+                }
+            } else {
+                LOGGER.debugCr(reconciliation, "Rolling Pod {} due to {}", pod.getMetadata().getName(), podRollReasons.getReasons());
+                return podRollReasons;
+            }
```

Comment:
So, does this need to be updated to handle both reasons = both annotations? Otherwise, why are we patching the clients CA generation in the PodSet?